### PR TITLE
Implement filtering and infinite loading to MPE Search Engine

### PR DESCRIPTION
**Pull request by Devessier**
Closes https://github.com/AdonisEnProvence/MusicRoom/issues/343
Closes https://github.com/AdonisEnProvence/MusicRoom/issues/342

Part of https://github.com/AdonisEnProvence/MusicRoom/issues/341

### CODE REVIEW

File: packages/server/app/Controllers/Http/MpeRoomsHttpController.ts [hunk 1]
```diff
@@ -1,45 +1,105 @@
...
+        const mpeRoomsPagination = await user
+            .related('mpeRooms')
+            .query()
+            .where('name', 'ilike', `${searchQuery}%`)
+            .orderBy([
+                {
+                    column: 'mpe_rooms.is_open',
+                    order: 'asc',
+                },
+                // FIXME: need to handle invitations
```
Comment (prastoin) [inline packages/server/app/Controllers/Http/MpeRoomsHttpController.ts:78]: Do we want the personal user rooms to be ordered by potential invitation ?

File: packages/client/tests/data/index.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export const db = factory({
 
     searchableMpeRooms: {
         roomID: primaryKey(() => datatype.uuid()),
-        roomName: () => random.words(datatype.number({ min: 1, max: 5 })),
+        roomName: () => random.words(),
```
Comment (prastoin) [inline packages/client/tests/data/index.ts:33]: :thumbsup:

Comment (prastoin) [reply packages/client/tests/data/index.ts:33]: should we be doing the same for the same file line 25 ?

File: packages/client/services/MpeService.ts [hunk 3]
```diff
@@ -1,41 +1,54 @@
...
-    });
-    const parsedResponse = LibraryMpeRoomSearchResponseBody.parse(
-        rawResponse.data,
-    );
+        searchQuery,
+        page,
+    };
+
+    const rawResponse = await redaxios.post(url, body);
+    const parsedResponse = MpeSearchMyRoomsResponseBody.parse(rawResponse.data);
```
Comment (prastoin) [inline packages/client/services/MpeService.ts:30]: This is a real pleasure to red
So few lines for a lot of work

**Review by prastoin — APPROVED** [review]
Looks great !
Hail to the state machines

File: packages/server/app/Controllers/Http/MpeRoomsHttpController.ts (hunk 1 above)
Comment (Devessier) [reply packages/server/app/Controllers/Http/MpeRoomsHttpController.ts:78]: I checked our Basecamp documentation, and it seems that we don't, indeed.
Do we even want to filter by opening status?

File: packages/client/tests/data/index.ts (hunk 2 above)
Comment (Devessier) [reply packages/client/tests/data/index.ts:33]: Why not! In another PR?

File: packages/client/services/MpeService.ts (hunk 3 above)
Comment (Devessier) [reply packages/client/services/MpeService.ts:30]: Haha yeah! Love it too.

File: packages/server/app/Controllers/Http/MpeRoomsHttpController.ts (hunk 1 above)
Comment (prastoin) [reply packages/server/app/Controllers/Http/MpeRoomsHttpController.ts:78]: Mhm I don't think so

**Pull request merged**